File: lib/search/binary_search.rb
module Search
  def self.binary_search(arr, value)
    low = 0
    high = arr.size - 1
  
    while low <= high
      mid = low + (high - low) / 2
  
      return true if arr[mid] == value
  
      if arr[mid] < value
        low = mid + 1
      else
        high = mid - 1
      end
    end
  
    return false
  end
end

File: public/javascripts/search/binary_search.js
var Search = {
  binarySearch (arr, value) {
    var low = 0,
      high = arr.length,
      mid;

    while(low <= high) {
      mid = Math.floor(low + (high - low) / 2);

      if (arr[mid] === value) return true;
      
      if (arr[mid] < value) {
        low = mid + 1;
      } else {
        high = mid - 1;
      }
    }

    return false;
  }
}